### DISCUSSION TITLE
Module Not Found

### DISCUSSION THREAD

**Discussion by kaisiant** [sofn-xyz/mailing#308] [Q&A] [answered]
Is it only me having this problem? 

<img width="483" alt="image" src="https://user-images.githubusercontent.com/26655111/199780053-fa5a3ce1-3ea4-422a-ab26-292cc105a2fc.png">

**Comment by alexfarrill** [comment]
@kaisian97 I haven't seen that one before.  Can you give me a bit more context about when you see this happen and what your environment is like (what platform, etc.)?

**Reply by kaisiant** [reply]
Can take a look at this repo https://github.com/kaisian97/mailing-bug

Basically, it is just a plain typescript next app.
1. yarn create next-app --typescript
2. yarn add mailing mailing-core

**Comment by alexfarrill** [comment]
Thanks @kaisian97, I am able to repro the problem using your repo.  I've also tried the latest (canary) version of next `13.0.3-canary.1` and mailing `0.8.14` but they have the same problem.  Let me know if you have any luck resolving, I will try a few more things as well

**Comment by alexfarrill** [comment]
@kaisian97 after fiddling with it for a bit, I was able to get it running by doing `yarn add fs-extra`.  not sure if that would work for you, but it might be a workaround for now if you try it?

**Comment by alexfarrill** [comment, marked as answer]
I'm actually not sure if that was what did it, I also uninstalled and reinstalled node (16.16.0) and npm (8.11.0) and now I can run this project without changes.  N.B. when you run `npx mailing dev` you should see the preview server start on :3883 not on :3000.  That might be related to the problem.  I'm not sure where 3000 came from since I don't see it in your mailing.config.json here.  It could be some bad interaction between this version of next and mailing?

**Reply by kaisiant** [reply]
The mailing preview server works fine. The actual problem is the buildSendMail function.

**Reply by alexfarrill** [reply]
@kaisian97 Ah, I see what is going on.  `sendMail` is being invoked from the client side in `pages/index.tsx`.  That is not going to work.  What you want to do is use sendMail in an api route or somewhere server side and then you could call the api route from client side if you want to trigger an email to be sent from something like a user event.  Does that make sense?  Hopefully that helps you get it working

**Reply by psugihara** [reply]
To elaborate a bit... the reason this doesn't work is because if sendMail was callable from the client, the client would need credentials for your SMTP transport. An attacker could easily grab those and use them to send any spam they want (and run up your SMTP bill!).
